Automattic/Gridicons-iOS
# Use PDF assets instead of auto-generated drawing code

This PR updates Gridicons to use PDF assets with vector data instead of auto-generated drawing code. The previous system was awkward to update and required people to use third party Mac software to update the library. The new system uses an asset catalog containing PDFs straight from the [main Gridicons repo](https://github.com/automattic/gridicons), and [SwiftGen](https://github.com/SwiftGen/SwiftGen) to generate an enum referencing the assets in the catalog.

I know this is a huge changeset, but it's primarily because of the removal of the old autogenerated code, and the addition of the new Xcode-created asset catalog.

I would like to bump the pod version to 1.0 when these changes are released, but I'll do that separately as it'll require updating other pods that depend on a 0.x version of Gridicons.

**To test**

* Ensure the framework builds and the tests pass
* Run `rake dependencies` and ensure SwiftGen is installed successfully
* Run `rake gen` and ensure that there are no major changes from the current `GridiconsGenerated.swift` file (you could try clearing that file first and ensure it's regenerated correctly).
* Ensure the demo project runs and you can view and resize the icons
* Check out the associated [WPiOS branch](https://github.com/wordpress-mobile/WordPress-iOS/tree/feature/gridicons-1.0). Build and run it, and ensure that icons show up in the app. Run through the main sections of the app ensuring everything looks as you'd expect.
